src/tests/ftest/control/dmg_network_scan.py

```diff
@@ -4,273 +4,72 @@
-        """ Get list of devices."""
-        devs = {}
-        for dev in os.listdir("/sys/class/net/"):
-            devs[dev] = [dev]
-            # Check if we need to add infiniband dev to list
-            p = "/sys/class/net/{}/device/infiniband".format(dev)
-            if os.path.exists(p):
-                devs[dev].extend(os.listdir(p))
-        return devs
-
-    @staticmethod
-    def clean_info(net_devs):
-        """Clean up the devices found in the system.
-
-        Clean up devices that don't have a numa node and some providers that
-        might be duplicates, also need to attach 'ofi' to provider name.
-
-        Args:
-            net_devs (NetDev): list of NetDev objects.
-
-        Returns:
-            NetDev: list of NetDev objects.
-
-        """
-        idx_l = []
-        for idx, dev in enumerate(net_devs):
-            if dev.f_iface == "lo":
-                idx_l.append(idx)
-            elif not dev.numa:
-                idx_l.append(idx)
-            else:
-                if dev.providers:
-                    no_dups = list(dict.fromkeys(dev.providers))
-                    dev.providers = ["ofi+" + i for i in no_dups]
-        _ = [net_devs.pop(idx - i) for i, idx in enumerate(sorted(idx_l))]
-
-        return net_devs
 
     def get_sys_info(self):
         """Get the system device information.
 
         Returns:
-            list: list of NetDev objects.
+            list: list of NetworkDevice objects.
 
         """
-        sys_net_devs = []
-
-        # Get device names on this system
-        dev_names = self.get_devs()
-        for dev in dev_names:
-            dev_info = NetDev(self.hostlist_servers[-1], dev)
-            for dev_name in dev_names[dev]:
-                dev_info.set_dev_info(dev_name, self.ofi_prefix)
-            sys_net_devs.append(dev_info)
-
-        # Create NetDev per provider to match dmg output
-        f_net_devs = []
-        for dev in self.clean_info(sys_net_devs):
-            if dev.providers:
-                for provider in dev.providers:
-                    new_dev = dev.get_copy()
-                    new_dev.providers = [provider]
-                    f_net_devs.append(new_dev)
-
-        # Get the provider specified in server config
-        cfg_p = self.server_managers[0].get_config_value("provider")
-        if cfg_p:
-            if cfg_p == "ofi+tcp":
-                cfg_p = "ofi+tcp;ofi_rxm"
-            elif cfg_p == "ofi+verbs":
-                cfg_p = "ofi+verbs;ofi_rxm"
-            f_net_devs = [dev for dev in f_net_devs if dev.providers == [cfg_p]]
-
-        return f_net_devs
+        server_provider = self.server_managers[0].get_config_value("provider")
+        sys_info = []
+        for entry in get_network_information(self.hostlist_servers, SUPPORTED_PROVIDERS):
+            if entry.device.startswith("ib") and server_provider in entry.provider:
+                sys_info.append(entry)
+                sys_info[-1].ib_device = None
```

Comment:
Optional - Personally, I think this is more clear
```suggestion
                entry.ib_device = None
                sys_info.append(entry)
```

Reply:
Done